src/main/kotlin/app/revanced/patches/youtube/layout/hide/getpremium/resource/patch/HideGetPremiumVideoAdvertisementResourcePatch.kt

```diff
@@ -0,0 +1,32 @@
+package app.revanced.patches.youtube.layout.hide.getpremium.resource.patch
+
+import app.revanced.patcher.annotation.Name
+import app.revanced.patcher.annotation.Version
+import app.revanced.patcher.data.ResourceContext
+import app.revanced.patcher.patch.PatchResult
+import app.revanced.patcher.patch.PatchResultSuccess
+import app.revanced.patcher.patch.ResourcePatch
+import app.revanced.patcher.patch.annotations.DependsOn
+import app.revanced.patches.shared.mapping.misc.patch.ResourceMappingPatch
+import app.revanced.patches.shared.settings.preference.impl.StringResource
+import app.revanced.patches.shared.settings.preference.impl.SwitchPreference
+import app.revanced.patches.youtube.layout.hide.getpremium.annotations.HideGetPremiumVideoAdvertisementCompatibility
+import app.revanced.patches.youtube.misc.settings.bytecode.patch.SettingsPatch
+
+@Name("hide-get-premium-resource-patch")
+@DependsOn([SettingsPatch::class])
+class HideGetPremiumVideoAdvertisementResourcePatch : ResourcePatch {
```

Comment:
You can also remove this patch and move the code in the `execute` method to the bytecode patch or:
```suggestion
@DependsOn([SettingsPatch::class])
class HideGetPremiumResourcePatch : ResourcePatch {
```